### PR TITLE
[task] use invoke context run instead of subprocess

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Use invoke `context.run` instead of subprocess to run commands from a script and fix locks cleanup.

### Motivation

Pulumi recently changed the way they name stacks, prefixing them with organisation and project. This broke our clean up script.

Then for some reason we could not figure yet out, using subprocess to run `pulumi stack ls --all` was returning a different result compared to the one we were getting running `pulumi` on terminal. Same binary, same version 🤯 

Switching to `invoke.ctx.run` fixed this, we added `pty=True` to always use a pseudo terminal, but it worked even without. Magic !

All in all, this PR fixes the cleanup task.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/new_e2e_tests.py [hunk 1]
```diff
@@ -148,44 +148,40 @@ def _clean_locks():
...
         for filename in os.listdir(Path(subdir)):
-            file_path = os.path.join(subdir, filename)
-            if os.path.isfile(file_path) and filename.endswith(".json"):
-                os.remove(file_path)
-                print(f"🗑️ Deleted lock: {file_path}")
+            path = os.path.join(subdir, filename)
+            if os.path.isfile(path) and filename.endswith(".json"):
+                os.remove(path)
+                print(f"🗑️ Deleted lock: {path}")
+            elif os.path.isdir(path):
```
Comment (ogaca-dd) [inline tasks/new_e2e_tests.py:155]: Can you explain why you added this case?

File: tasks/new_e2e_tests.py [hunk 2]
```diff
@@ -148,44 +148,40 @@ def _clean_locks():
...
-    subprocess.call(
-        [
-            "pulumi",
-            "stack",
-            "rm",
-            "--force",
-            stack_name,
-        ]
-    )
+    with ctx.cd('/tmp'):
```
Comment (ogaca-dd) [inline tasks/new_e2e_tests.py:170]: Why not using `tempfile.gettempdir()`?

File: tasks/new_e2e_tests.py [hunk 3]
```diff
@@ -148,44 +148,40 @@ def _clean_locks():
...
+        lines = lines[1:]  # skip headers
+        e2e_stacks: List[str] = []
+        for line in lines:
+            stack_name = line.split(" ")[0]
+            e2e_stacks.append(stack_name)
+        return e2e_stacks
+
+
+def _remove_stack(ctx: Context, stack_name: str):
+    ctx.run(f"pulumi stack rm --force --yes --stack {stack_name}", pty=True)
```
Comment (ogaca-dd) [inline tasks/new_e2e_tests.py:184]: 👍  for `--yes`.

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: tasks/new_e2e_tests.py (hunk 2 above)
Comment (pducolin) [reply tasks/new_e2e_tests.py:170]: Actually better yes, compatible with Windows too. Changing this.

File: tasks/new_e2e_tests.py (hunk 1 above)
Comment (pducolin) [reply tasks/new_e2e_tests.py:155]: Pulumi recently changed the way the name stack adding organisation and project, this has impact also on how locks are stored. Before this change, we were looking at `.json` file on root `locks` folder as they were all there. Now locks are in `.locks/<organisation>/<project>/<stack>.json`. I could change this to recursively look for `json` file and remove only them, but it seems to be safe to remove recursively all subfolders directly.

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**